### ISSUE TITLE
Optimize the setting of sync start block in light client mode

### ISSUE THREAD

**Issue by Danie0918** [Magickbase/neuron-public-issues#304]
![image](https://github.com/Magickbase/neuron-public-issues/assets/122437870/24fa88cd-b459-42f0-bb7f-773c7101ae44)
![image](https://github.com/Magickbase/neuron-public-issues/assets/122437870/6d3c599d-2aa0-493d-9940-18f1cb02b624)

In the current light client mode, the user can set the synchronization start block. When the block number set by the user is after the historical transaction block number, then the historical transaction will not be synchronized, but displaying 100% synchronization can be misleading to the user.

This needs to be optimized:

1.Add a strong hint when setting the start block number.
2.Consider optimizing the setting of the start block number range not to exceed the latest block number.
3.Add a reminder that the start block number has been set.

**Comment by Danie0918** [comment]
[PRD](https://github.com/Magickbase/neuron-public-issues/pull/227#issuecomment-1774620449)

【Optimize the setting of sync start block in light client mode】

- Requirement Description:When the block number set by the user is after the historical transaction block number, then the historical transaction will not be synchronized, but displaying 100% synchronization can be misleading to the user.


![optimize_the_setting_of_sync_start_block_in_light_client_mode](https://github.com/Magickbase/neuron-public-issues/assets/122437870/658c7684-31a4-46ce-b2e3-4b93e433b969)


- Requirement Contents
1. Add a strong prompt when setting the starting block number and require 5 seconds before clicking to confirm the operation.
2. When the set block number is larger than the latest block it errors and changes the set block number to the latest block number, and it takes 5s to confirm.
3. Add manually set starting block in the Synchronization dropdown box to support opening the corresponding block in the Explorer.

**Comment by yanguoyu** [comment]
1. <img width="408" alt="image" src="https://github.com/Magickbase/neuron-public-issues/assets/12881040/8cb5d62a-9c96-45c0-93d6-7106e84a7073">

Is it a wrong description? Does it should be `Set start block number`? Or do you mean it will clear all synchronized data and start from the setting start block number?

2. > When the set block number is larger than the latest block it errors and changes the set block number to the latest block number, and it takes 5s to confirm.

It notifies that `unable to set a higher block number`, should we disable the confirm button? Then it is unnecessary to take 5s to confirm. Another problem is that the latest block number will increase about every 10s, so should we refresh the latest block number and check whether the number is bigger every 10s

**Comment by Danie0918** [comment]
> Is it a wrong description? Does it should be `Set start block number`? Or do you mean it will clear all synchronized data and start from the setting start block number?

I'll change it. It's still set synchronization start block.

> It notifies that `unable to set a higher block number`, should we disable the confirm button? Then it is unnecessary to take 5s to confirm. Another problem is that the latest block number will increase about every 10s, so should we refresh the latest block number and check whether the number is bigger every 10s

Of course, it would be better to just disable it, but here is a scenario that considers the inability to get the latest blocks in real time.

It doesn't need to be very real-time here, refreshing is unnecessary.

**Comment by yanguoyu** [comment]
I suggest clearing all synchronized data and starting from the setting start block number if the set start block number is smaller than the last setting(or synced block number). When users set a smaller setting, it will not sync from the smaller setting because the light client has synced to a higher block number.
Should we notify the user or default execute that?

**Comment by Danie0918** [comment]
> I suggest clearing all synchronized data and starting from the setting start block number if the set start block number is smaller than the last setting(or synced block number). When users set a smaller setting, it will not sync from the smaller setting because the light client has synced to a higher block number. Should we notify the user or default execute that?

Notifying the user would be better, I'll add a popup alert later.

**Comment by Danie0918** [comment]
UI：https://www.figma.com/file/6XNoimRDbFTTNm016rbIdU/Magickbase?type=design&node-id=28438-26974&mode=design&t=AY5lLSGCmoHUwzpQ-0

**Comment by Keith-CY** [comment]
Another note could be added that if there are no transactions on the chain, or users assure there are no transactions on the chain, the tip block number could be set.

This note comes from the case that users find empty transaction lists on the explorer and they don't know how to continue.

**Comment by Danie0918** [comment]
> Another note could be added that if there are no transactions on the chain, or users assure there are no transactions on the chain, the tip block number could be set.
> 
> This note comes from the case that users find empty transaction lists on the explorer and they don't know how to continue.

Is this prompted when setting the start block?But for new wallets, we are not prompted to set the start block.

**Comment by Keith-CY** [comment]
> > Another note could be added that if there are no transactions on the chain, or users assure there are no transactions on the chain, the tip block number could be set.
> > This note comes from the case that users find empty transaction lists on the explorer and they don't know how to continue.
> 
> Is this prompted when setting the start block?But for new wallets, we are not prompted to set the start block.

Yes, it should be noted when setting the start point, but it's not the case of a freshly generated wallet(and a new wallet doesn't need the step of specifying the start point because it will be done automatically).

It's the case that a wallet was created before but never used. So the user cannot find the earliest transaction on ckb explorer when the wallet is imported.

**Comment by Danie0918** [comment]
> Another note could be added that if there are no transactions on the chain, or users assure there are no transactions on the chain, the tip block number could be set.
> 
> This note comes from the case that users find empty transaction lists on the explorer and they don't know how to continue.

![image](https://github.com/Magickbase/neuron-public-issues/assets/122437870/16b92141-5a43-4e76-8b13-f1bfb30461f4)

In order to be compatible with the case of setting the starting block for wallets with no transaction history, we have divided the case of setting the starting block into 2 cases:

1. Wallet with transaction history ： `Please note that transactions before the synchronization start block will not be synchronized！`
2. Wallets without transaction history ： `No transaction history found, you can start synchronizing from the latest block or any known block. `

What do you think about this? @Keith-CY @yanguoyu

**Comment by Keith-CY** [comment]
> > Another note could be added that if there are no transactions on the chain, or users assure there are no transactions on the chain, the tip block number could be set.
> > This note comes from the case that users find empty transaction lists on the explorer and they don't know how to continue.
> 
> ![image](https://private-user-images.githubusercontent.com/122437870/287183790-16b92141-5a43-4e76-8b13-f1bfb30461f4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE1MDgzNDEsIm5iZiI6MTcwMTUwODA0MSwicGF0aCI6Ii8xMjI0Mzc4NzAvMjg3MTgzNzkwLTE2YjkyMTQxLTVhNDMtNGU3Ni04YjEzLWYxYmZiMzA0NjFmNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMjAyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIwMlQwOTA3MjFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02ZmNjODVjNGRiYTFiMzQwNDUwMzQ4MTkyM2Q0MjdkMmYxNDgxZWRkOTBjOWZkODhkNDUyOGZmZGZhYWY3ZDRmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.jM2pebl5BZaSBgvC8olGbBE3pK3X8qfnJCDYkp0Tm_w)
> 
> In order to be compatible with the case of setting the starting block for wallets with no transaction history, we have divided the case of setting the starting block into 2 cases:
> 
> 1. Wallet with transaction history ： `Please note that transactions before the synchronization start block will not be synchronized！`
> 2. Wallets without transaction history ： `No transaction history found, you can start synchronizing from the latest block or any known block. `
> 
> What do you think about this? @Keith-CY @yanguoyu

I'm afraid these two messages should be displayed at the same time because Neuron cannot get the transaction count before the synchronization.

**Comment by yanguoyu** [comment]
https://github.com/nervosnetwork/neuron/pull/2994

**Comment by silySuper** [comment]
verified
